spec.emu

```diff
@@ -1114,6 +1141,969 @@
+                <th>Field Name</th>
+                <th>Value</th>
+                <th>Meaning</th>
+              </tr>
+            </thead>
+            <tr>
+              <td>[[From]]</td>
+              <td>a Position Record</td>
+              <td>Use [[Binding]] from this position until either the next [[From]] or the ranges' [[End]] position to retrieve this variables' value.</td>
+            </tr>
+            <tr>
+              <td>[[Binding]]</td>
+              <td>a String or *null*</td>
+              <td>The expression to use to retrieve this variables' value, or *null* if the variable is unavailable.</td>
+            </tr>
+          </table>
+        </emu-table>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-scopes-grammar">
+      <h1>Scopes grammar</h1>
+      <p>The scopes field is described by the following <em>context-sensitive</em> grammar starting at the |Scopes| goal symbol:</p>
+      <emu-grammar type="definition">
+        Scopes :
+          OriginalScopeTreeList
+          TopLevelItemList
+          OriginalScopeTreeList `,` TopLevelItemList
+
+        OriginalScopeTreeList :
+          OriginalScopeTreeItem
+          OriginalScopeTreeList `,` OriginalScopeTreeItem
+
+        OriginalScopeTreeItem :
+          OriginalScopeTree
+          [empty]
+
+        TopLevelItemList :
+          TopLevelItem
+          TopLevelItemList `,` TopLevelItem
+
+        TopLevelItem :
+          GeneratedRangeTree
+          UnknownItem
+      </emu-grammar>
+
+      <p>The number of |OriginalScopeTreeItem|s in the |OriginalScopeTreeList| is expected to match the length of the sources field. The n-th |OriginalScopeTreeItem| describes the scope tree of the n-th authored source file in the sources field. If no scope information is available for a particular source URL, an empty item can be used.</p>
+
+      <emu-note example>
+        <p>A source map with three authored sources where only the first and third source file have scope information available would be encoded as:</p>
+        <pre>
+          "scopes": "&lt;OriginalScopeTree&gt;,,&lt;OriginalScopeTree&gt;,&lt;GeneratedRangeTree&gt;"
+        </pre>
+      </emu-note>
+
+      <emu-clause id="sec-original-scope-grammar">
+        <h1>Original scope grammar</h1>
+        <p>Each original scope consists of:</p>
+        <ul>
+          <li>a |OriginalScopeStart| item describing the start location and general scope information.</li>
+          <li>an optional |OriginalScopeVariablesItem| describing the original names of the variables this scope declares.</li>
+          <li>an optional list of |OriginalScopeTree|s describing child scopes.</li>
+          <li>a |OriginalScopeEnd| item describing the end location.</li>
+        </ul>
+        <p>In grammar form:</p>
+        <emu-grammar type="definition">
+          OriginalScopeTree :
+            OriginalScopeStart OriginalScopeVariablesItem? OriginalScopeItemList? `,` OriginalScopeEnd
```

Comment:
Very slight preference here for doing ``OriginalScopeStart `,` OriginalScopeVariablesItem? OriginalScopeItemList? OriginalScopeEnd`` and giving `OriginalScopeVariablesItem` and `OriginalScopeItemList` a trailing rather than leading comma, so that the disambiguation for whether they are present or not is on the first character and not on the second.